### DISCUSSION TITLE
CoreML and Metal

### DISCUSSION THREAD

**Discussion by dubefab** [ggml-org/whisper.cpp#2092] [Q&A]
Hey! I'm pretty satisfied with whisper.cpp leveraging Metal on my m1 pro.
I am curious: if I build and use the CoreML model: will it speed it up even further?
Or is one or the other (pick between Metal or CoreML)? Not together?

**Comment by jingcodeguy** [comment]
@dubefab I think you could launch a test for different building settings. For instance, I have tried to build with CoreML and CoreML + Embed the Metal Library. I use `time` command for performance reference with the same audio file, limit to 1 process to benchmark the test. I haven't done too much consecutive tests. 

While it is not a serious or big benchmark test, based on the result, I found that
- with embedding, takes around 57s to complete
- wihtout embedding, takes around 220 to complete
The length of the audio is 42 minutes, Traditional Chinese

Just for reference.
